src/utils/components/SSHSecretSection/SSHSecretSection.tsx

```diff
@@ -1,52 +1,36 @@
-import React, { Dispatch, FC, SetStateAction, useEffect, useState } from 'react';
+import React, { FC, useState } from 'react';
 
+import { useInstanceTypeVMStore } from '@catalog/CreateFromInstanceTypes/zustand-state/useInstanceTypeVMStore';
 import { modelToGroupVersionKind, SecretModel } from '@kubevirt-ui/kubevirt-api/console';
 import { IoK8sApiCoreV1Secret } from '@kubevirt-ui/kubevirt-api/kubernetes';
-import SecretDropdown from '@kubevirt-utils/components/SSHSecretSection/utils/components/SecretDropdown/SecretDropdown';
-import SecretSelectionRadioGroup from '@kubevirt-utils/components/SSHSecretSection/utils/components/SecretSelectionRadioGroup';
-import SSHKeyUpload from '@kubevirt-utils/components/SSHSecretSection/utils/components/SSHKeyUpload/SSHKeyUpload';
-import {
-  SecretSelectionOption,
-  SSHSecretDetails,
-} from '@kubevirt-utils/components/SSHSecretSection/utils/types';
+import SecretDropdown from '@kubevirt-utils/components/SSHSecretSection/components/SecretDropdown/SecretDropdown';
+import SecretSelectionRadioGroup from '@kubevirt-utils/components/SSHSecretSection/components/SecretSelectionRadioGroup';
+import SSHKeyUpload from '@kubevirt-utils/components/SSHSecretSection/components/SSHKeyUpload/SSHKeyUpload';
+import { SecretSelectionOption } from '@kubevirt-utils/components/SSHSecretSection/utils/types';
 import { ALL_NAMESPACES_SESSION_KEY } from '@kubevirt-utils/hooks/constants';
-import { t } from '@kubevirt-utils/hooks/useKubevirtTranslation';
+import { useKubevirtTranslation } from '@kubevirt-utils/hooks/useKubevirtTranslation';
 import { useK8sWatchResource } from '@openshift-console/dynamic-plugin-sdk';
-import { useActiveNamespace } from '@openshift-console/dynamic-plugin-sdk-internal';
 import { Grid, GridItem, PopoverPosition } from '@patternfly/react-core';
 
 import HelpTextIcon from '../HelpTextIcon/HelpTextIcon';
 
 import './SSHSecretSection.scss';
 
-type SSHSecretSectionProps = {
-  sshSecretDetails: SSHSecretDetails;
-  setSSHSecretDetails: Dispatch<SetStateAction<SSHSecretDetails>>;
-};
-
-const SSHSecretSection: FC<SSHSecretSectionProps> = ({ sshSecretDetails, setSSHSecretDetails }) => {
-  const [activeNamespace] = useActiveNamespace();
+const SSHSecretSection: FC = () => {
+  const { t } = useKubevirtTranslation();
+  const { activeNamespace } = useInstanceTypeVMStore();
   const [secretSelectionOption, setSecretSelectionOption] = useState<SecretSelectionOption>(
     SecretSelectionOption.none,
   );
 
-  const secretsResourceData = useK8sWatchResource<IoK8sApiCoreV1Secret[]>({
+  const [secrets, ...loadedAndErrorData] = useK8sWatchResource<IoK8sApiCoreV1Secret[]>({
```

Comment:
Secrets are used anywhere else? I see it drilling and have a store. if it's used more than once maybe add it to store

Reply:
Is it really prop drilling if it goes only one component down? it's used only here so I think adding to the store is not really useful in this case

Reply:
yes, that's what I was asking, so if no drilling all good.